hathor/p2p/states/hello.py

```diff
@@ -168,3 +169,16 @@ def handle_hello(self, payload: str) -> None:
             return
 
         protocol.change_state(protocol.PeerState.PEER_ID)
+
+
+def _parse_sync_versions(hello_data: Dict[str, Any]) -> Set[SyncVersion]:
+    """Versions that are not recognized will not be included."""
+    if settings.CAPABILITY_SYNC_VERSION in hello_data['capabilities']:
+        if 'sync_versions' not in hello_data:
+            raise HathorError('protocol error, expected sync_versions field')
+        known_values = set(x.value for x in SyncVersion)
+        recognized_values = set(hello_data['sync_versions']) & known_values
+        return set(SyncVersion(x) for x in recognized_values)
+    else:
+        # XXX: implied value when sync-version capability isn't present
+        return {SyncVersion.V1}
```

Comment:
It seems we should return `{SyncVersion.V1} & known_values`. I mean, my code doesn't work because they have different types, but we are filtering in one case and not filtering here. It seems like a bug.

Reply:
I don't understand. That'd be redundant because `SyncVersion.V1` is always "known", by definition. So even if we made `known_values = set(SyncVersion)` (which would skip the SyncVersion -> str -> SyncVersion), it wouldn't make any difference because `SyncVersion.V1 in SyncVersion` is _always_ `True`.

Reply:
I got it. Should it be in the `SyncVersion` enum?

Reply:
Maybe, it makes sense to be a classmethod there.

Reply:
Now that I'm looking the whole diff, I think it is good here. It needs the `hello_data` and is specific to this step of the connection.